test/e2e/node/stress_probe.go

```diff
@@ -0,0 +1,200 @@
+		livenessProbe := v1.Probe{
+			ProbeHandler: v1.ProbeHandler{
+				GRPC: &v1.GRPCAction{
+					Port:    containerPort,
+					Service: nil,
+				},
+			},
+			InitialDelaySeconds: probeTestInitialDelaySeconds,
+			TimeoutSeconds:      5, // default 1s can be pretty aggressive in CI environments with low resources
+			FailureThreshold:    1,
+		}
+		pod := livenessPodSpec("grpc", f.Namespace.Name, livenessProbe)
+		runLivenessTest(ctx, f, pod)
+	})
+})
+
+func livenessPodSpec(probeType, namespace string, livenessProbe v1.Probe) *v1.Pod {
+	pod := e2epod.NewAgnhostPod(namespace, "liveness-stress-"+string(uuid.NewUUID()), nil, nil, nil)
+	pod.ObjectMeta.Labels = map[string]string{"test": "liveness-stress"}
+	pod.Spec.Containers[0].LivenessProbe = &livenessProbe
+	pod.Spec.Containers = append(pod.Spec.Containers, make([]v1.Container, numContainers-1)...)
+	for idx := 0; idx < numContainers; idx++ {
+		cc := containerConfig{
+			Name:         "agnhost-container-" + strconv.Itoa(idx+1),
+			ProbeType:    probeType,
+			ContainerIdx: idx,
+		}
+		instance := *pod.Spec.Containers[0].DeepCopy()
+		pod.Spec.Containers[idx] = newContainer(cc, instance)
+	}
+	return pod
+}
+
+func newContainer(config containerConfig, instance v1.Container) v1.Container {
+	newContainerPort := config.ContainerIdx + containerPort + 1
+	instance.Name = config.Name
+	instance.Ports = []v1.ContainerPort{{ContainerPort: int32(newContainerPort)}}
+
+	switch config.ProbeType {
+	case "tcp":
+		instance.LivenessProbe.TCPSocket.Port = intstr.FromInt(newContainerPort)
+		instance.Args = []string{"serve-hostname", "--tcp", "--port", strconv.Itoa(newContainerPort), "--http=false"}
+	case "http":
+		instance.LivenessProbe.HTTPGet.Port = intstr.FromInt(newContainerPort)
+		instance.Args = []string{"serve-hostname", "--http", "--port", strconv.Itoa(newContainerPort)}
+	case "grpc":
+		instance.LivenessProbe.GRPC.Port = int32(newContainerPort)
+		instance.Args = []string{"grpc-health-checking", "--port", strconv.Itoa(newContainerPort), "--http-port", strconv.Itoa(newContainerPort + containerPort)}
+	}
+
+	return instance
+}
+
+func httpGetHandler(path string, port int) v1.ProbeHandler {
+	return v1.ProbeHandler{
+		HTTPGet: &v1.HTTPGetAction{
+			Path: path,
+			Port: intstr.FromInt(port),
+		},
+	}
+}
+
+func tcpSocketHandler(port int) v1.ProbeHandler {
+	return v1.ProbeHandler{
+		TCPSocket: &v1.TCPSocketAction{
+			Port: intstr.FromInt(port),
+		},
+	}
+}
+
+func getRestartCount(p *v1.Pod) int {
+	count := 0
+	for _, containerStatus := range p.Status.ContainerStatuses {
+		count += int(containerStatus.RestartCount)
+	}
+	return count
+}
+
+// runLivenessTest verifies the number of restarts.
+func runLivenessTest(ctx context.Context, f *framework.Framework, pod *v1.Pod) {
+	podClient := e2epod.NewPodClient(f)
+	ns := f.Namespace.Name
+	gomega.Expect(pod.Spec.Containers).NotTo(gomega.BeEmpty())
```

Comment:
I'm not sure we need to check anything here, but if we do it probably makes sense to expect pod to have `NumContainers` containers.

Reply:
changed, please check this one too.